ibm/data_source_ibm_satellite_host_script.go

```diff
@@ -0,0 +1,131 @@
+package ibm
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	homedir "github.com/mitchellh/go-homedir"
+	"github.ibm.com/ibmcloud/kubernetesservice-go-sdk/kubernetesserviceapiv1"
+)
+
+func dataSourceIBMSatelliteAttachHostScript() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIBMSatelliteAttachHostScriptRead,
+
+		Schema: map[string]*schema.Schema{
+			"location": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "A unique name for the new Satellite location",
+			},
+			"labels": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Set:         schema.HashString,
+				Description: "List of labels for the attach host",
+			},
+			"host_provider": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"script_dir": {
+				Description: "The directory where the satellite attach host script to be downloaded. Default is home directory ",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"host_script": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Attach host script content",
+			},
+		},
+	}
+}
+
+func dataSourceIBMSatelliteAttachHostScriptRead(d *schema.ResourceData, meta interface{}) error {
+	location := d.Get("location").(string)
+	hostProvider := d.Get("host_provider").(string)
+	scriptDir := d.Get("script_dir").(string)
+
+	satClient, err := meta.(ClientSession).SatelliteClientSession()
+	if err != nil {
+		return err
+	}
+
+	bxClient, err := meta.(ClientSession).BluemixSession()
+	if err != nil {
+		return err
+	}
+
+	getSatLocOptions := &kubernetesserviceapiv1.GetSatelliteLocationOptions{
+		Controller: &location,
+	}
+
+	locData, _, err := satClient.GetSatelliteLocation(getSatLocOptions)
+	if err != nil {
+		return err
+	}
+
+	// script labels
+	labels := make(map[string]string)
+	if _, ok := d.GetOk("labels"); ok {
+		l := d.Get("labels").(*schema.Set)
+		labels = flattenHostLabels(l.List())
```

Comment:
No need to get multiple times..
```
if v, ok := d.GetOk("labels"); ok {
		l := v.(*schema.Set)
```
should be fine

Reply:
Updated code.